### PR TITLE
169376952 parse sheets 4a 5a misc prices

### PR DESCRIPTION
**Pull request by gidjin**
## Description

Add the parsing methods to parse and save in temp tables the pricing data from sheets 4a and 5a of the pricing spreadsheet.

## Reviewer Notes

4a is for Shipment Management Services Prices, Counseling Services Prices, and Transition Prices. 5a has Domestic and International move accessorial prices as well as additional prices. Parsing the data for these sheets will generate 6 files.

I ran `make gofmt` to make sure my changes were formatted according to the go style rules and it changed a handful of other unrelated files which I also committed.

## Setup

To run the tests:

```sh
go test ./pkg/parser/pricing -v
```

To build and run the command-line tool:
```sh
rm -f bin/ghc-pricing-parser
make  bin/ghc-pricing-parser
ghc-pricing-parser --filename pkg/parser/pricing/fixtures/pricing_template_2019-09-19_fake-data.xlsx --save --display --xlsxSheets "16,17"
```

## Code Review Verification Steps

* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/tree/master/docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/tree/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [#169376952](https://www.pivotaltracker.com/story/show/169376952) for this change

### CODE REVIEW

File: pkg/parser/pricing/parse_access_and_add_prices_test.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+		XlsxFilename: suite.xlsxFilename,
+		XlsxSheets:   []string{strconv.Itoa(sheetIndex)},
+		SaveToFile:   true,
+		RunTime:      time.Now(),
+		XlsxFile:     suite.xlsxFile,
+		RunVerify:    true,
+	}
+
+	err := verifyAccessAndAddPrices(params, sheetIndex)
+	// TODO check errormessage
```
Comment (reggieriser) [inline pkg/parser/pricing/parse_access_and_add_prices_test.go]: Did you mean to leave this TODO in or go ahead and check the error message?

File: pkg/parser/pricing/parse_access_and_add_prices_test.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+	slice, err := parseDomesticInternationalAdditionalPrices(params, sheetIndex)
+	suite.NoError(err, "parseDomesticInternationalAdditionalPrices function failed")
+
+	outputFilename := dataSheet.generateOutputFilename(sheetIndex, params.RunTime, swag.String("additional"))
+	err = createCSV(outputFilename, slice)
+	suite.NoError(err, "could not create CSV")
+
+	const goldenFilename string = "17_5a_access_and_add_prices_additional_golden.csv"
+	suite.helperTestExpectedFileOutput(goldenFilename, outputFilename)
+}
```
Comment (reggieriser) [inline pkg/parser/pricing/parse_access_and_add_prices_test.go]: You could have a test that checks the wrong sheet functionality in the parse methods to mirror the test you have for verify, but that's not critical.

File: pkg/parser/pricing/parse_management_counseling_transition_prices_test.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+		XlsxFilename: suite.xlsxFilename,
+		XlsxSheets:   []string{strconv.Itoa(sheetIndex)},
+		SaveToFile:   true,
+		RunTime:      time.Now(),
+		XlsxFile:     suite.xlsxFile,
+		RunVerify:    true,
+	}
+
+	err := verifyManagementCounselTransitionPrices(params, sheetIndex)
+	// TODO check errormessage
```
Comment (reggieriser) [inline pkg/parser/pricing/parse_management_counseling_transition_prices_test.go]: Some thing here on the TODO -- just do the check?

File: pkg/parser/pricing/parse_management_counseling_transition_prices_test.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	slice, err := parseTransitionPrices(params, sheetIndex)
+	suite.NoError(err, "parseTransitionPrices function failed")
+
+	outputFilename := dataSheet.generateOutputFilename(sheetIndex, params.RunTime, swag.String("transition"))
+	err = createCSV(outputFilename, slice)
+	suite.NoError(err, "could not create CSV")
+
+	const goldenFilename string = "16_4a_mgmt_coun_trans_prices_transition_golden.csv"
+	suite.helperTestExpectedFileOutput(goldenFilename, outputFilename)
+}
```
Comment (reggieriser) [inline pkg/parser/pricing/parse_management_counseling_transition_prices_test.go]: Same thing here about a possible check for wrong sheet on the parse methods like you have on the verify method.

File: pkg/parser/pricing/parse_access_and_add_prices_test.go (hunk 1 above)
Comment (gidjin) [reply pkg/parser/pricing/parse_access_and_add_prices_test.go]: I did intend to do that

File: pkg/parser/pricing/parse_management_counseling_transition_prices_test.go (hunk 3 above)
Comment (gidjin) [reply pkg/parser/pricing/parse_management_counseling_transition_prices_test.go]: Yes, I did. Originally I had the note so I could come back after looking up how to do the check.

File: pkg/parser/pricing/parse_access_and_add_prices_test.go (hunk 2 above)
Comment (gidjin) [reply pkg/parser/pricing/parse_access_and_add_prices_test.go]: Makes sense to me, I'll add a test.

File: pkg/parser/pricing/parse_management_counseling_transition_prices_test.go (hunk 4 above)
Comment (gidjin) [reply pkg/parser/pricing/parse_management_counseling_transition_prices_test.go]: Will do

**Pull request merged**